job_creator/test/test_queue_consumer.py

```diff
@@ -1,95 +1,113 @@
-# pylint: disable=missing-module-docstring, missing-class-docstring, missing-function-docstring, protected-access,
-# pylint: disable=too-many-instance-attributes
-import unittest
 from unittest import mock
 
+import pytest
 from pika import PlainCredentials
 
 from jobcreator.queue_consumer import QueueConsumer
 
+MESSAGE_CALLBACK = mock.MagicMock()
+QUEUE_HOST = mock.MagicMock()
+USERNAME = mock.MagicMock()
+PASSWORD = mock.MagicMock()
+QUEUE_NAME = mock.MagicMock()
 
-class QueueConsumerTest(unittest.TestCase):
-    @mock.patch("jobcreator.queue_consumer.BlockingConnection")
-    @mock.patch("jobcreator.queue_consumer.ConnectionParameters")
-    def setUp(self, connection_parameters, blocking_connection):
-        self.blocking_connection = blocking_connection
-        self.connection_parameters = connection_parameters
-        self.message_callback = mock.MagicMock()
-        self.queue_host = mock.MagicMock()
-        self.username = mock.MagicMock()
-        self.password = mock.MagicMock()
-        self.queue_name = mock.MagicMock()
 
-        self.quc = QueueConsumer(self.message_callback, self.queue_host, self.username, self.password, self.queue_name)
+@pytest.fixture()
+def setup_queue_consumer():
+    with (
+        mock.patch("jobcreator.queue_consumer.ConnectionParameters") as connection_parameters,
+        mock.patch("jobcreator.queue_consumer.BlockingConnection") as blocking_connection,
+    ):
+        quc = QueueConsumer(MESSAGE_CALLBACK, QUEUE_HOST, USERNAME, PASSWORD, QUEUE_NAME)
+        return quc, blocking_connection, connection_parameters
 
-    def test_init_creates_credentials_and_connection_parameters(self):
-        self.assertEqual(self.message_callback, self.quc.message_callback)
-        self.assertEqual(self.queue_host, self.quc.queue_host)
-        self.assertEqual(self.queue_name, self.quc.queue_name)
 
-        credentials = PlainCredentials(username=self.username, password=self.password)
-        self.connection_parameters.assert_called_once_with(self.queue_host, 5672, credentials=credentials)
-        self.assertEqual(self.connection_parameters.return_value, self.quc.connection_parameters)
+@pytest.mark.usefixtures("setup_queue_consumer")
```

Comment:
Can you cut all these lines out if you mark the fixture as `pytest.fixture(autouse=True, scope="module"`)

Something like that?

Reply:
Nice piece of tech! thanks for the suggestion, implemented!